sketch.js

```diff
@@ -150,53 +149,92 @@ function showRhymesOnly() {
   }
 }
 
+function showAllGroups() {
+  for (var w = 0; w < words.length; w++) {
+    var word = words[w];
+    if (/\r|\n/.exec(word)) span = createSpan("<br/>");
+    else span = createSpan(word);
+    span.parent(output);
+    if (!/\W+/.test(word)) {
+      for (var g = 0; g < groups.length; g++) {
+        var group = groups[g];
+        if (group.indexOf(word) != -1) {
+          var ci = groups.indexOf(group);
+          span.style('background-color', colors[ci]);
+        }
+      }
+    }
+  }
+}
+
+function findGroup(cRhymes, wColor) {
+  var foundGroup = false;
+  // Going backwards, loop through all groups
+  for (var j = groups.length-1; j >= 0; j--) {
+    var group = groups[j];
+    if (cRhymes.indexOf(group[0]) != -1) {
+      // It rhymed, put it in that group and use that color
+      foundGroup = true;
+      groups[j].push(words[index]);
+      wColor = colors[j];
+    }
+  }
+  if (!foundGroup) {
+    // We didn't find any rhymes, make it its own group and own color
+    usecolor = bcolor;
+    groups.push([words[index]])
+    colors.push(wColor);
+  }
+}
+
 // Goes through each term, waiting for the previous to be finished
 function analyzeWords() {
   // We'll do this until we get to the last term
   if (index < words.length) {
-    // Only look for rhymes if it's a word, not a space
-    if (!/\W+/.test(words[index])) {
-      // Start an async request with an anonymous callback
-      httpGetAsync('http://rhymebrain.com/talk?function=getRhymes&word=' + words[index],function(data){
-        span = createSpan(words[index]);
-        span.parent(output);
-        var cRhymes = [];
+    // Only look for rhymes if it's a word, not a space or empty string
+    if (!/\W+/.test(words[index]) && !(words[index] === "")) {
+      var cRhymes = [];
+      var wColor = color(random(100,240), random(100,240), random(100,240));
+      // create UI for the results as they roll in
+      var span = createSpan(words[index]);
+      span.parent(output);
 
-        // Take all rhymes with an accuracy score over 200
-        for (var k = 0; k < data.length; k++) {
-          if (data[k]["score"] > 200) cRhymes.push(data[k]["word"]);
+      // First looks through our previous RhymeBrain responses to see if we already have the answer
+      for (var k = 0; k < rhymingWordLists.length; k++){
```

Comment:
I'm getting the following error when I run this in my browser:
<img width="568" alt="Screen Shot 2020-02-19 at 8 40 41 PM" src="https://user-images.githubusercontent.com/3926534/74892714-667e4480-5358-11ea-9e7a-26f798854790.png">

Reply:
Is this just missing a `var rhymingWordLists =[];` at the top of the file?

Reply:
Oops, I have it locally but it looks like I missed it when I picked my changes. I'll push up the fix shortly

Reply:
Alright, should be all good now